src/Microsoft.DotNet.Wpf/tests/UnitTests/PresentationFramework.Fluent.Tests/ControlTests/BaseControlTests.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using PresentationFramework.Fluent.Tests.TestUtilities;
+using System.Windows.Controls;
+
+
+namespace PresentationFramework.Fluent.Tests.ControlTests;
+
+public class BaseControlTests : IDisposable
+{
+    public BaseControlTests()
+    {
+        TestWindow = SetupTestWindow();
+    }
+
+    public virtual List<FrameworkElement> GetStyleParts(Control element)
+    {
+        return new List<FrameworkElement>();
+    }
+
+    public virtual void VerifyControlProperties(FrameworkElement element, ResourceDictionary expectedProperties)
+    {
+        return;
+    }
+
+    #region Protected Methods
+
+    protected void AddControlToView(Window window, FrameworkElement fe)
+    {
+        ArgumentNullException.ThrowIfNull(window);
+
+        if (window.Content is not StackPanel panel)
+            throw new ArgumentException($"The window does not have StackPanel as it's root element.");
+        
+        panel.Children.Add(fe);
+    }
+
+    protected void RemoveControlFromView(Window window, FrameworkElement fe)
+    {
+        if (window is null)
+            return;
+
+        if (window.Content is not StackPanel panel)
+            return;
+
+        panel.Children?.Remove(fe);
+    }
+
+    protected void ClearView(Window window)
+    {
+        if (window is null)
+            return;
+
+        if (window.Content is not StackPanel panel)
+            return;
+
+        panel.Children?.Clear();
+    }
+
+    protected Window SetupTestWindow()
+    {
+        Window window = new Window() { Width = 800, Height = 600 };
+        
+        StackPanel sp = new StackPanel() { 
+            Name = "RootPanel",
+            HorizontalAlignment = HorizontalAlignment.Stretch,
+            Orientation = Orientation.Vertical
+        };
+
+        window.Content = sp;
+
+        return window;
+    }
+
+    protected Window SetupTestWindow(ColorMode mode)
+    {
+        Window window = new Window() { Width = 800, Height = 600 };
+        StackPanel sp = new StackPanel() { Name = "RootPanel" };
+        SetColorMode(window, mode);
+        window.Content = sp;
```

Comment:
I noticed that you are setting `window.ThemeMode` to `ThemeMode.None` in `ResetTestWindow`, but you are not setting it in `SetupTestWindow` to some valid value. Additionally, you are accepting the mode as an argument and adding the current window to the `TestWindows` map corresponding to the `mode`, but the `mode` is not being set in the window object.